vulcanai/datasets/tabulardataset.py

```diff
@@ -473,10 +475,49 @@ def identify_low_variance(self, threshold):
                     dct_low_var[col] = col_var
         return dct_low_var
 
+    def convert_all_categorical_binary(self, list_only=False,
```

Comment:
@nirtiac shouldn't this be easily extendable to n>2 categorical types? similar to how the cat codes works in pandas? or is this specifically avoided?

Reply:
yeah, that's specifically avoided, we have functions for both categorical and one-hot encoding. One-hot encoding would create two columns, whereas this only creates one. (speaking of, there's a difference between dummy coding and one-hot encoding, where one would create 3 columns for 3 categories and the other 2, as the 3rd column can be 'inferred'. We right now create 3, which is more understandable but somewhat out of line with how we're doing these binary columns) .

Reply:
yeah I see. perhaps there may be some utility iim being able to preserve the one additional column and create dummy codes for n>2?

Reply:
right now that's what we do for n>2

Reply:
perfect. thanks for the clarification.